autoextract/apikey.py

```diff
@@ -11,10 +11,14 @@ class NoApiKey(Exception):
 
 def get_apikey(key: Optional[str] = None) -> str:
     """ Return API key, probably loading it from an environment variable """
-    if key is not None:
-        return key
-    try:
-        return os.environ[ENV_VARIABLE]
-    except KeyError:
+    if not key:
```

Comment:
This is a behavior change - it was intentional that empty string is different from not passing a key. Do you have this issue because of Scrapy integration, where it is hard to set a setting to None from the command line?

Reply:
> it was intentional that empty string is different from not passing a key

What's the desired outcome with this behavior?

> Do you have this issue because of Scrapy integration, where it is hard to set a setting to None from the command line?

Actually, no. Tests have failed on my machine because I had this variable set and I thought about "fixing" this blank string problem while creating a fixture that patched system environment during tests.

I don't understand why it would be desired to have a blank string returned from the get_apikey() method since it's an invalid value and will culminate in a failed request. But if you'd like me to I can just revert the logic and keep focused on the test fixture to avoid failing because of my own environment.

Reply:
Thanks for the explanation @victor-torres! I'm fine with this change, but we need to make sure we mention it in the changelog as technically backwards incompatible.